src/js/templatedComponent.js

```diff
@@ -0,0 +1,201 @@
+    // syntax and other objects, resolve only the strings using stringTemplate
+    // against the set of terms.
+    sjrk.storyTelling.templatedComponent.resolveTerms = function (terms) {
+        return fluid.transform(terms, function (term) {
+            // TODO: Make this functionality recursive
+            // if (typeof term === "array" || typeof term === "object") {
+            //     return sjrk.storyTelling.templatedComponent.resolveTerms(term);
+            // } else {
+            if (typeof term === "string") {
+                return fluid.stringTemplate(term, terms);
+            } else {
+                return term;
+            }
+            // }
+        });
+    };
+
+    // Adds gpii-binder to bind form components and models
+    fluid.defaults("sjrk.storyTelling.templatedComponentWithBinder", {
+        gradeNames: ["gpii.binder", "sjrk.storyTelling.templatedComponent"],
+        // Specified by using grade to bind form markup
+        // to component model;
+        // see https://github.com/GPII/gpii-binder
+        bindings: {
+        },
+        events: {
+            "onBindingApplied": null
+        },
+        // Applies bindings from gpii-binder after
+        // the template is loaded
+        listeners: {
+            "onTemplateRendered.applyBinding": {
+                funcName: "gpii.binder.applyBinding",
+                args: "{that}"
+            },
+            "onTemplateRendered.fireOnBindingApplied": {
+                func: "{that}.events.onBindingApplied.fire",
+                priority: "after:applyBinding"
+            }
+        }
+    });
+
+    // TODO: refer to Tony's work for further implementation
+    // https://wiki.gpii.net/w/Technology_Evaluation_-_Internationalising_and_Localising_UI_strings#Detailed_Review
+    fluid.defaults("sjrk.storyTelling.templatedComponentWithLocalization", {
+        gradeNames: ["sjrk.storyTelling.templatedComponent"],
+        listeners: {
+            "{resourceLoader}.events.onResourcesLoaded": {
+                "func": "sjrk.storyTelling.messageLoaderWithLocalization.loadLocalizationMessages",
+                "args": ["{resourceLoader}.resources.componentMessages.resourceText", "{that}", "templateTerms"],
+                "priority": "before:renderTemplateOnSelf",
+                "namespace": "loadLocalizationMessages"
+            }
+        },
+        components: {
+            resourceLoader: {
+                type: "sjrk.storyTelling.messageLoaderWithLocalization"
+            }
+        }
+    });
+
+    fluid.defaults("sjrk.storyTelling.messageLoader", {
+        gradeNames: ["fluid.resourceLoader"],
+        resources: {
+            // Specified by using grade
+            // The template file using the
+            // fluid.stringTemplate syntax
+            // componentTemplate: ""
+        },
+        terms: {
+            "resourcePrefix": "."
+        }
+    });
+
+    fluid.defaults("sjrk.storyTelling.messageLoaderWithLocalization", {
```

Comment:
I add to and correct what I wrote above: the `storyAuthoring.handlebars` file is actually loaded with localization search. While it is loaded with a `sjrk.storyTelling.messageLoader` (rather than `messageLoaderWithLocalization`), the `sjrk.storyTelling.storyAuthoring` grade has a `distributeOptions` that applies a `locale`:

    resourceLoaderConfig: {
        resourcePrefix: ".",
        locale: "en"
    },
    distributeOptions: [{
        source: "{that}.options.resourceLoaderConfig.resourcePrefix",
        target: "{that resourceLoader}.options.terms.resourcePrefix"
    },
    {
        source: "{that}.options.resourceLoaderConfig.locale",
        target: "{that resourceLoader}.options.locale"
    }],

If I load the top level `index.html` in Firefox and look at the Network developer tab, I see:

- 200 `/src/templates/storyAuthoring.handlebars`
- 404 `/src/templates/storyAuthoring_en.handlebars`
- 200 `/src/templates/storyEdit.handlebars`
- 404 `/src/messages/storyMessages.json`
- 404 `/src/templates/storyEdit_en.handlebars`
- 200 `/src/messages/storyMessages_en.json`